paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/ProjectionRowData.java

```diff
@@ -0,0 +1,250 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink;
+
+import org.apache.paimon.types.RowType;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.table.data.ArrayData;
+import org.apache.flink.table.data.DecimalData;
+import org.apache.flink.table.data.MapData;
+import org.apache.flink.table.data.RawValueData;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.StringData;
+import org.apache.flink.table.data.TimestampData;
+import org.apache.flink.types.RowKind;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+
+/**
+ * A {@link RowData} that provides a mapping view of the original {@link RowData} according to
+ * projection information.
+ */
+public class ProjectionRowData implements RowData, Serializable {
```

Comment:
ProjectedRowData?

Reply:
ProjectedRowData has already existed in Flink and also in Paimon, which only supports non-nested projection and is now used in Kafka CDC. Its implementation is different from that of the current ProjectionRowData, so I prefer we decide on the implementation and then seek if the two classes could merge into one.